src/renderer/setupPasswordOptional.html

```diff
@@ -0,0 +1,304 @@
+          </div>
+
+          <!-- Loading message -->
+          <div class="splash-loading-message hidden" id="activity">Setting up Holochain...</div>
+        </div>
+        <div class="splash-license">Licensed under the Cryptographic Autonomy License v1.0</div>
+      </div>
+      <img
+        style="position: absolute; bottom: 5px; right: 10px; width: 300px"
+        src="img/powered_by_holochain.png"
+      />
+      <img
+        tabindex="0"
+        id="close-btn"
+        title="close"
+        style="cursor: pointer; position: absolute; top: 5px; right: 5px; width: 35px"
+        src="img/close.svg"
+      />
+    </div>
+    <script>
+      // TODO add event listener for holochain fatal panic case
+
+      let password = '';
+      let confirmedPassword = '';
+
+      (async () => {
+        const { productName, version } = await window.electronAPI.getNameAndVersion();
+        document.getElementById('productName').innerHTML = productName;
+        document.getElementById('version').innerHTML = `version ${version}`;
+        document.getElementById(
+          'choose-setup-sentence'
+        ).innerHTML = `Choose whether you want to set up ${productName} with or without a password.`;
+      })();
+
+      const activitEl = document.getElementById('activity');
+      window.electronAPI.onProgressUpdate((event, msg) => {
+        console.log('status', msg);
+        activitEl.innerHTML = msg;
+      });
+      const closeBtn = document.getElementById('close-btn');
+      closeBtn.addEventListener('click', () => {
+        window.electronAPI.exit();
+      });
+      closeBtn.addEventListener('keypress', (e) => {
+        if (e.key === 'Enter' || e.key === ' ') {
+          window.electronAPI.exit();
+        }
+      });
+
+      const chooseSetupDiv = document.getElementById('choose-setup-div');
+      const withPwBtn = document.getElementById('with-password-btn');
+      const withoutPwBtn = document.getElementById('without-password-btn');
+      const warning0El = document.getElementById('warning-0');
+
+      const choosePwDiv = document.getElementById('choose-password-div');
+      const choosePwInput = document.getElementById('choose-password-input');
+      const continueBtn = document.getElementById('continue-btn');
+      const warning1El = document.getElementById('warning-1');
+      const previousStepBtn0 = document.getElementById('previous-step-btn-0');
+
+      const confirmPwDiv = document.getElementById('confirm-password-div');
+      const confirmPwInput = document.getElementById('confirm-password-input');
+      const launchBtn = document.getElementById('launch-btn');
+      const warning2El = document.getElementById('warning-2');
+      const previousStepBtn = document.getElementById('previous-step-btn');
+      const understoodDiv = document.getElementById('understood-div');
+      const understoodCheckbox = document.getElementById('understood-checkbox');
+
+      const handleLaunch = async () => {
```

Comment:
This code seems to be duplicated, perhaps you can derive a common js module you can import in the respective html files that need it

Reply:
I don't think that would be particularly more ergonomic in this case and potentially add possibilities to mistakes. They are each tied to local html elements and js variables so that function would need to take as arguments references to the respective html elements and js variables which would again make it quite unwieldy. And I think I prefer to keep that logic directly in the html file to make sure that things don't get mixed up if the html elements change in any of the files because they don't necessarily the same UI and html elements may change independently.